rflx/expression.py

```diff
@@ -416,20 +436,24 @@ def symbol(self) -> str:
 
 class LogExpr(AssExpr):
     def __str__(self) -> str:
-        if not self.terms:
-            return str(TRUE)
-        result = ""
-        for i, t in reversed(list(enumerate(self.terms))):
-            if i == 0:
-                result = self.parenthesized(t) + result
-            else:
-                result = (
-                    "\n"
-                    + str(self.symbol)[1:]
-                    + indent_next(self.parenthesized(t), len(self.symbol) - 1)
-                    + result
-                )
-        return result
+        try:
+            return self.__str
+        except AttributeError:
+            if not self.terms:
+                return str(TRUE)
```

Comment:
Shouldn't this be stored in `self.__str` then? Otherwise we get `AttributeError` on every `str` invocation?

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/390/commits/92983bdc4d9c64b48d67b21c245d402eb83d0ba9.